src/components/atoms/Input.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+interface Props {
+    type: string,
+    className: string,
+    setState: React.Dispatch<React.SetStateAction<string>>
+}
+
+const Input: React.FC<Props> = ({type, className, setState}) => {
+
+    return (
+
+        <input type={type} className={className} onChange={(e) => setState(e.target.value)} />
```

Comment:
du coup içi se sera 

```suggestion
        <input type={type} className={className} onChange={(e) =>onChange(e.target.value)} />
```

et tu fais ton setState au composant au dessus 😁